### PR TITLE
Fix PlayUIServer.detach() to allow sequential attach-detach of StatsStorage

### PR DESCRIPTION
**Pull request by printomi**
## What changes were proposed in this pull request?

Fix https://github.com/deeplearning4j/deeplearning4j/issues/6859
This is an improvement upon commit https://github.com/deeplearning4j/deeplearning4j/pull/6858/commits/d2c0df6876a6bd5441e82c226513f05759f84b5d (part of PR https://github.com/deeplearning4j/deeplearning4j/pull/6858 )
Added ND4J backend for TestPlayUI (new test scoped dependency for deeplearning4j-play).

## How was this patch tested?

I wrote a new unit test for the sequential case (`testUISequentialSessions`), and run it manually in IntelliJ IDEA.

### CODE REVIEW

**Comment by printomi** [comment]
Sorry for the large number of commits in this PR, the last 6 commit would have been enough. I guess, I should have branched from printomi:deeplearning4j/master branch?

**Comment by printomi** [comment]
I managed create a clean branch with only 4 commits, and so a new PR https://github.com/deeplearning4j/deeplearning4j/pull/6950 instead of this PR. The discussion can continue there.
